R/app.R

```diff
@@ -1,167 +1,215 @@
+        
+        tags$div(
+          id="gbif-block",
+          class="data-block text-input-block",
+          tags$label("for"="gbif_name", "Enter a taxon name to load points from GBIF:"),
+          tags$div(
+            "class" = "text-input-wrapper",
+            textInput("gbif_name", label=NULL)
+          ),
+          tags$span(id="gbif-hint", class = "hint", "e.g. Cyphostemma njegerre"),
+          actionButton("queryGBIF", "Load points"),
+          tags$span(
+            "class" = "toggle-wrapper",
+            shinyjs::hidden(
+              ## GBIF points on/off ----
+              shinyWidgets::prettySwitch(
+                inputId = "gbif_onoff",
+                label = "GBIF",
+                value = TRUE,
+                status = "success",
+                fill = TRUE
+              )
             )
+          )
+        ),
+        
+        tags$div(
+          id="powo-block",
+          class="data-block text-input-block",
+          tags$label("for"="powo_id", "Enter a POWO ID for a native range map:"),
+          tags$div(
+            "class" = "text-input-wrapper",
+            textInput("powo_id", label=NULL)
           ),
-        )
-      ),
-      
-      br(),
-      
-      shiny::fluidRow(
-        ## CSV input widget ----
-        column(
-          12,
-          align = "centre",
-          shiny::helpText("Upload a CSV with at least 'longitude', 'latitude' fields "),
-          shiny::fileInput(
-            "csv_in",
-            NULL,
-            multiple = FALSE,
-            accept = (".csv")
+          tags$span(
+            "id" = "powo-hint",
+            "class" = "hint",
+            "e.g. 68179-1. Search",
+            tags$a(href="https://powo.science.kew.org/", target="_blank", "POWO"),
+            "to get accepted name ID."
           ),
-        ),
-      ),
-      ## GBIF input field ----
-      shiny::fluidRow(column(
-        12, align = "center", verbatimTextOutput("gbifValidation")
-      )),
-      
-      fluidRow(
-        column(12, align="left",
-               tags$h5("Enter a taxon name to load points from GBIF:")
-        )
-      ),
-      
-      fluidRow(
-        column(8, align="center",
-               textInput("gbif_name", label=NULL, placeholder="Cyphostemma njegerre")),
-        column(4, align="center",
-               actionButton("queryGBIF", "load points"))
-      ),
-      ## POWO ID field ----
-      fluidRow(
-        column(12, align="left",
-               tags$h5("Enter a POWO ID for a native range map:")
+          actionButton("queryPOWO", "Load map"),
+          tags$span(
+            "class" = "toggle-wrapper",
+            shinyjs::hidden(
+              shinyWidgets::prettySwitch(
+                inputId = "native_onoff",
+                label = "Exclude non-native",
+                value = FALSE,
+                status = "danger",
+                fill = TRUE
+              )
+            )
+          )
         )
       ),
```

Comment:
Confirm this the intended tab order?

![image](https://github.com/stevenpbachman/shinygeocat/assets/932250/c15607f4-eb1c-41e0-a576-c29ef55cf1d4)

Reply:
It actually is because it needs to make sense on narrower screens where the load button is placed below.

![Screenshot from 2023-09-07 13-30-36](https://github.com/stevenpbachman/shinygeocat/assets/7906373/46678082-3a81-4701-99d3-94760fe52db2)